nav2_collision_monitor/src/source.cpp

```diff
@@ -75,6 +75,10 @@ bool Source::sourceValid(
   const rclcpp::Time & source_time,
   const rclcpp::Time & curr_time) const
 {
+  // Ignore timestamp check if source_timeout == 0.0
+  if (source_timeout_.seconds() == 0.0) {
+    return true;
+  }
```

Comment:
If source is outdated too much, `spourceValid()` will return true and we still try to interpolate sources on current transform time. I think, it is more correct to interpolate the sources if time shift is between 0 ... and some `source_timeout`, and ignore outdated sources, if time shift is lager than `source_timeout` (if developer want this). Otherwise, we could get some unpredictable failures (e.g. what will be, if some data source is not available within 5 min?)

So, I consider to have `ignore_invalid` option, or using negative timeouts for ignored sources.

Reply:
Changed the logic in last push